serveradmin/serverdb/query_committer.py

```diff
@@ -435,18 +435,18 @@ def _validate_regexp(changes, servers, servertype_attributes):
         server = servers[object_id]
         for attribute_id, change in attribute_changes.items():
             sa = servertype_attributes[server['servertype']].get(attribute_id)
-            if not sa or not sa.regexp:
+            if not sa or not sa.attribute or not sa.attribute.regexp:
                 continue
 
             action = change['action']
             if action == 'update' or action == 'new':
                 if change['new'] is None:
                     continue
-                if not sa.regexp_match(change['new']):
+                if not sa.attribute.regexp_match(change['new']):
```

Comment:
Accessing `sa.attribute` would cause a database query.  We should use the objects we already looked up.